CPP_STYLE_GUIDE.md

```diff
@@ -257,6 +258,27 @@ class ExampleClass {
 };
 ```
 
+### When to use AliasedBuffers
+
+When composite objects, specifically JS arrays (that have backing
+native buffers), are operated globally (modifiable outside of the
+lexical scope) or out-of-band (modifiable outside of the callsite)
+the observability of such changes are lost from Virtual Machine layer.
+So if the object's backing data is being modified from C++ side
+out-of-band with the calling context, the recommended abstraction to use
+is AliasedBuffer class, that has place holders to hook observers
+for data mutations.
```

Comment:
This paragraph is very hard to read for me.

I don’t want to tell you what to write, but I think something like this would do:

> When modifying typed arrays directly from C++, use an `AliasedBuffer` when possible,
> because that makes it easier for some JS debuggers to observe these modifications.

We’d probably want to go into more detail when – if ever – something like node-chakracore gets merged into Node.js core, and we actually need to take care of always making modifications observable. For now, the chakracore team has been doing that on their own in the places where we don’t use `AliasedBuffer`.

Reply:
@addaleax - thanks, pushed a modified version with **almost** your suggested wordings. ptal, thanks.